File: examenes-Andres/Tr1Final/Ejercicio_4/Coche.py
from Vehiculo import Vehiculo


class Coche(Vehiculo):
    ventanilla: str

    def __init__(self, nombre, dorsal, velocidad_maxima, tipo_combustible, numero_ruedas):
        super().__init__(nombre, dorsal, velocidad_maxima, tipo_combustible, numero_ruedas)
        self.ventanilla = "subida"

    def bajar_ventanilla(self):
        self.ventanilla = "bajada"
        print(f"La ventanilla del coche con dorsal {self.dorsal} esta bajada")

    def mostrar_info(self):
        if self.ventanilla == "bajada":
            return "El coche con " + super().mostrar_info() + f", tiene la ventanilla {self.ventanilla}"
        return "El coche con " + super().mostrar_info()
    

File: Ejercicios-examen-final1ev/src/Clases/Coche.java
package Clases;

public class Coche extends Vehiculo {
    private String ventanilla;

    public Coche(String nombre, int dorsal, int velocidadActual, int velocidadMaxima, String tipoCombustible, int numeroRuedas, String ventanilla) {
        super(nombre, dorsal, velocidadActual, velocidadMaxima, tipoCombustible, numeroRuedas);
        this.ventanilla = ventanilla;
    }

    public Coche(String ventanilla) {
        this.ventanilla = ventanilla;
    }

    @Override
    public String toString() {
        String superstring = super.toString();
        return superstring + ", ventanilla: " + ventanilla + "\n";
    }

    public String subirBajarVentanilla() {
        if (this.ventanilla == "bajada") {
            return this.ventanilla = "subida";
        } else {
            return this.ventanilla = "bajada";
        }
    }

    public String getVentanilla() {
        return ventanilla;
    }

    public void setVentanilla(String ventanilla) {
        this.ventanilla = ventanilla;
    }
}
